src/saltext/vmware/states/vmc_networks.py

```diff
@@ -0,0 +1,474 @@
+                subnets=subnets,
+                admin_state=admin_state,
+                description=description,
+                domain_name=domain_name,
+                tags=tags,
+                advanced_config=advanced_config,
+                l2_extension=l2_extension,
+                dhcp_config_path=dhcp_config_path,
+                display_name=display_name,
+            )
+
+            if "error" in updated_network:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_network["error"], result=False
+                )
+
+            updated_network = __salt__["vmc_networks.get_by_id"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                network_id=network_id,
+                verify_ssl=verify_ssl,
+                cert=cert,
+            )
+
+            if "error" in updated_network:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_network["error"], result=False
+                )
+
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Updated Network {}".format(network_id),
+                old_state=network,
+                new_state=updated_network,
+                result=True,
+            )
+        else:
+            log.info("All fields are same as existing Network %s", network_id)
+            return vmc_state._create_state_response(
+                name=name, comment="Network exists already, no action to perform", result=True
+            )
+    else:
+        log.info("No Network found with Id %s", network_id)
+        created_network = __salt__["vmc_networks.create"](
+            hostname=hostname,
+            refresh_key=refresh_key,
+            authorization_host=authorization_host,
+            org_id=org_id,
+            sddc_id=sddc_id,
+            network_id=network_id,
+            verify_ssl=verify_ssl,
+            cert=cert,
+            subnets=subnets,
+            admin_state=admin_state,
+            description=description,
+            domain_name=domain_name,
+            tags=tags,
+            advanced_config=advanced_config,
+            l2_extension=l2_extension,
+            dhcp_config_path=dhcp_config_path,
+        )
+
+        if "error" in created_network:
+            return vmc_state._create_state_response(
+                name=name, comment=created_network["error"], result=False
+            )
+
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Created Network {}".format(network_id),
+            new_state=created_network,
+            result=True,
+        )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    network_id,
```

Comment:
**suggestion** remove ``network_id`` parameter and use ``name`` instead. Applies to both present and absent states.

Reply:
this change we have planned to do as part of https://github.com/saltstack/salt-ext-modules-vmware/issues/120 for all the modules of VMC at once.